adapter/internal/adapter/adapter.go

```diff
@@ -223,8 +224,24 @@ func Run(conf *config.Config) {
 	if eventHubEnabled {
 		// Load subscription data
 		eventhub.LoadSubscriptionData(conf)
+		var isAzureEventingFeatureFlagEnabled bool
+		var err error
+
+		envValue := os.Getenv("FEAURE_FLAG_REPLACE_EVENT_HUB")
+		if (envValue != "") {
+			isAzureEventingFeatureFlagEnabled, err = strconv.ParseBool(envValue)
+			if (err != nil) {
+				logger.LoggerMgw.Error("Error occurred while parsing FEAURE_FLAG_REPLACE_EVENT_HUB environment value.",
+					err)
+			}
+		}
 
-		go messaging.ProcessEvents(conf)
+		if (isAzureEventingFeatureFlagEnabled) {
+			logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting to integrate with azure service bus")
+			go messaging.InitiateAndProcessEvents(conf)
+		} else {
```

Comment:
This should not be an else, it needs to be done anyway. Because we don't need out feature to break the existing functionality.

Reply:
fixed